tests/boa/pool/test_RevenueGeneratingContract.py

```diff
@@ -337,14 +343,100 @@ def test_cant_claim_rev_of_non_pool_token(pool, admin, base_asset, amount):
     assert pool.claimable_rev(base_asset) == 0
     with boa.reverts("non-revenue token"):
         pool.claim_rev(base_asset, amount, sender=admin)
-    
+
     new_token = boa.load('tests/mocks/MockERC20.vy', "Lending Token", "LEND", 18)
     new_token.mint(pool, amount)
     assert new_token.balanceOf(pool) == amount
     assert pool.claimable_rev(new_token) == 0
     with boa.reverts("non-revenue token"):
         pool.claim_rev(new_token, amount, sender=admin)
 
+
+@pytest.mark.pool
+@pytest.mark.slow
+# @pytest.mark.rev_generator # What does this do?
+@given(amount=st.integers(min_value=1, max_value=MAX_UINT)) # min_val = 1 so no off by one when adjusting values
+@settings(max_examples=100, deadline=timedelta(seconds=1000))
+def test_self_owner_rev_claimable_by_rev_recipient(pool, admin, amount):
```

Comment:
We already have a test for what u wrote here in `test_rev_recipient_can_claim_rev`. I want to test that all RevenueGenerated events emitted with `self.owner` as the receiver are claimable by `rev_recipient`. So should randomly generate fees (any type or just the ones that pay to owner idc) and then test that claimable rev == event amounts

Reply:
done. rewrote the test and created a helper function called `gen_rev_events` that calls different functions (`mint`, `deposit`, `redeem`, `withdraw`, etc) that log the `RevenueGenerated` event.